pkg/resources/grant_ownership_test.go

```diff
@@ -0,0 +1,518 @@
+package resources
+
+import (
+	"testing"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetOnObjectIdentifier(t *testing.T) {
+	testCases := []struct {
+		Name       string
+		ObjectType sdk.ObjectType
+		ObjectName string
+		Expected   sdk.ObjectIdentifier
+		Error      string
+	}{
+		{
+			Name:       "database - account object identifier",
+			ObjectType: sdk.ObjectTypeDatabase,
+			ObjectName: "test_database",
+			Expected:   sdk.NewAccountObjectIdentifier("test_database"),
+		},
+		{
+			Name:       "database - account object identifier - quoted",
+			ObjectType: sdk.ObjectTypeDatabase,
+			ObjectName: "\"test_database\"",
+			Expected:   sdk.NewAccountObjectIdentifier("test_database"),
+		},
+		{
+			Name:       "schema - database object identifier",
+			ObjectType: sdk.ObjectTypeSchema,
+			ObjectName: "test_database.test_schema",
+			Expected:   sdk.NewDatabaseObjectIdentifier("test_database", "test_schema"),
+		},
+		{
+			Name:       "schema - database object identifier - quoted",
+			ObjectType: sdk.ObjectTypeSchema,
+			ObjectName: "\"test_database\".\"test_schema\"",
+			Expected:   sdk.NewDatabaseObjectIdentifier("test_database", "test_schema"),
+		},
+		{
+			Name:       "table - schema object identifier",
+			ObjectType: sdk.ObjectTypeTable,
+			ObjectName: "test_database.test_schema.test_table",
+			Expected:   sdk.NewSchemaObjectIdentifier("test_database", "test_schema", "test_table"),
+		},
+		{
+			Name:       "table - schema object identifier - quoted",
+			ObjectType: sdk.ObjectTypeTable,
+			ObjectName: "\"test_database\".\"test_schema\".\"test_table\"",
+			Expected:   sdk.NewSchemaObjectIdentifier("test_database", "test_schema", "test_table"),
+		},
+		{
+			Name:       "validation - valid identifier",
+			ObjectType: sdk.ObjectTypeDatabase,
+			ObjectName: "to.many.parts.in.this.identifier",
+			Error:      "unable to classify identifier",
+		},
+		{
+			Name:       "validation - unsupported type",
+			ObjectType: sdk.ObjectTypeShare,
+			ObjectName: "some_share",
+			Error:      "object_type SHARE is not supported",
+		},
+		{
+			Name:       "validation - invalid account object identifier",
+			ObjectType: sdk.ObjectTypeDatabase,
+			ObjectName: "test_database.test_schema",
```

Comment:
We recently had the discussion about the validations for account object identifiers that concluded that it should be less strict for account identifiers, right? What were the arrangements there? Is it a topic that we will revisit with identifiers rework or...?

Reply:
Yeah, that's my bad. This should be possible, so I'll change it and I'll add TODO to address this when we'll be working on identifiers.